python/ray/includes/gcs_client.pxd

```diff
@@ -0,0 +1,69 @@
+from libcpp cimport bool as c_bool
+from libcpp.memory cimport shared_ptr, unique_ptr
+from libcpp.string cimport string as c_string
+
+from libc.stdint cimport uint8_t, int32_t, uint64_t, int64_t
+from libcpp.unordered_map cimport unordered_map
+from libcpp.vector cimport vector as c_vector
+from libcpp.pair cimport pair as c_pair
+
+from ray.includes.common cimport (
+    CRayStatus,
+    CGcsClientOptions,
+)
+
+cdef extern from "ray/gcs/accessor.h" nogil:
+    cdef cppclass CInternalKVAccessor "ray::gcs::InternalKVAccessor":
+        CRayStatus Put(const c_string &key,
+                       const c_string &value,
+                       c_bool overwrite,
+                       c_bool &updated)
+        CRayStatus Get(const c_string &key, c_string &value)
+        CRayStatus Del(const c_string &key)
+        CRayStatus Exists(const c_string &key, c_bool &exist)
+        CRayStatus Keys(const c_string &key, c_vector[c_string]& results)
+
+cdef extern from "ray/gcs/gcs_client.h" nogil:
+    cdef cppclass CGcsClient "ray::gcs::GcsClient":
+        CInternalKVAccessor &InternalKV()
+
+
+cdef extern from * namespace "_gcs_maker":
+    """
```

Comment:
I thought about putting it in a cpp file, but then realized that it's only for python only.
So I think that maybe I should just put it here.

Reply:
+1 on a separate cpp file, but I don't think that should be a blocker for this being merged.

Reply:
Here is that this is more or less cython only code, so that's why I'm putting it here.
I'm not against it, but any comment on why do you prefer a separate cpp file?

Reply:
I think it's more of a preference to keep the cython layer thin because we don't have a lot of cython expertise, and we tend to treat it as a "glue" layer, so there's very little test coverage. 

> I don't think that should be a blocker for this being merged.

+1

Reply:
Got it 👍